visuals/lenss_plotter.py

```diff
@@ -0,0 +1,272 @@
+        file_path_eve = folder + self.evening_file
+
+        col_names = [
+            "Time (UTC)",
+            "Time (CST)",
+            "Temperature",
+            "Frequency",
+            "Voltage",
+            "Sensor",
+        ]
+        df_morn = pd.read_csv(
+            file_path_morn, low_memory=False, sep=";", names=col_names
+        )
+        df_eve = pd.read_csv(
+            file_path_eve, low_memory=False, sep=";", names=col_names
+        )
+
+        return df_morn, df_eve
+
+    def _parse_data(self, df):
+        """
+        This method cleans the data files and organizes it
+        with the relevant column names as a dataframe.
+        """
+
+        # Assuming the timestamp column is stored in a variable called
+        # "timestamp_str"
+        timestamp_format = (
+            "%Y-%m-%dT%H:%M:%S.%f"  # Define the format of the timestamp
+        )
+
+        df["Time (CST)"] = pd.to_datetime(
+            df["Time (CST)"], format="%Y-%m-%dT%H:%M:%S.%f"
+        )
+        df["Time (UTC)"] = pd.to_datetime(
+            df["Time (UTC)"], format="%Y-%m-%dT%H:%M:%S.%f"
+        )
+
+        df["Day"] = df["Time (UTC)"].dt.day
+        df["Month"] = df["Time (UTC)"].dt.month
+        df["Year"] = df["Time (UTC)"].dt.year
+        df["Date"] = df["Time (UTC)"].apply(lambda x: x.date())
+
+    def _construct_data(self):
+        """
+        This method constructs all the information needed for the
+        class to render the visualization.
+        """
+        pass
+
+    def _get_moon_info(self, df: pd.DataFrame):
+        def get_moon_illum(t: datetime):
+            return float(str(self.sensor_location.moon_illumination(str(t))))
+
+        df["moon_illum"] = df["Time (CST)"].apply(lambda t: get_moon_illum(t))
+
+    def _call_apis(self):
+        return super()._call_apis()
+
+    def _call_weather_api(self, t_start: datetime, t_end: datetime):
+        """
+        This method pulls the relevant weather for the night
+        from external APIs.
+
+        Returns: dictionary of hours, cloud cover percentages
+        """
+        api_url = (
+            f"https://archive-api.open-meteo.com/v1/archive"
+            f"?latitude={self.sensor_latitude}&"
+            f"longitude={self.sensor_longitude}"
+            f"&start_date={self.evening_day}"
+            f"&end_date={self.morning_day}&hourly=cloudcover"
+            f"&timezone=America%2FChicago"
+        )
+        response = requests.get(api_url)
+        data = response.json()
+
+        if not data:
+            raise Exception("Cloud Data Couldn't be Found")
+
+        times = [
+            datetime.strptime(time, "%Y-%m-%dT%H:%M%f")
+            for time in data["hourly"]["time"]
+        ]
+        cloudcover = data["hourly"]["cloudcover"]
+
+        cloud_dict = dict(zip(times, cloudcover))
+        return {x: cloud_dict[x] for x in cloud_dict if t_start <= x <= t_end}
+
+    def _get_sunset_sunrise(self, t_start: datetime, t_end: datetime):

```

Comment:
It looks like you're covering some outer scope variables with `t_start` and `t_end`.